ui/app/components/auth/form/base.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { service } from '@ember/service';
+import { task } from 'ember-concurrency';
+import { waitFor } from '@ember/test-waiters';
+
+import type AuthService from 'vault/vault/services/auth';
+import type ClusterModel from 'vault/models/cluster';
+import type { HTMLElementEvent } from 'vault/forms';
+
+/**
+ * @module Auth::Base
+ *
+ * @param {string} authType - chosen login method type
+ * @param {object} cluster - The cluster model which contains information such as cluster id, name and boolean for if the cluster is in standby
+ * @param {function} onError - callback if there is a login error
+ * @param {function} onSuccess - calls onAuthResponse in auth/page redirects if successful
+ */
+
+interface Args {
+  authType: string;
+  cluster: ClusterModel;
+  onError: CallableFunction;
+  onSuccess: CallableFunction;
+}
+
+export default class AuthBase extends Component<Args> {
+  @service declare readonly auth: AuthService;
+
+  @action
+  onSubmit(event: HTMLElementEvent<HTMLFormElement>) {
+    event.preventDefault();
+    const formData = new FormData(event.target as HTMLFormElement);
+    const data: Record<string, FormDataEntryValue | null> = {};
+
+    for (const key of formData.keys()) {
+      data[key] = formData.get(key);
+    }
+
+    this.login.unlinked().perform(data);
+  }
+
+  login = task(
+    waitFor(async (data) => {
+      try {
+        const authResponse = await this.auth.authenticate({
+          clusterId: this.args.cluster.id,
+          backend: this.args.authType,
+          data,
+          selectedAuth: this.args.authType,
+        });
+
+        // responsible for redirect after auth data is persisted
+        this.onSuccess(authResponse);
+      } catch (error) {
+        this.onError(error as Error);
+      }
+    })
+  );
+
+  // if we move auth service authSuccess method here (or to each auth method component)
+  // then call that before calling parent this.args.onSuccess
```

Comment:
As I mentioned in my comment above, I'm making changes as incrementally as possible. When this component is wired up it will be easier to strip the auth controller of unnecessary logic and move it to the relevant components.